modules/pool/lib/pool-onchain-data.ts

```diff
@@ -0,0 +1,211 @@
+    multicaller.call(`${id}.totalSupply`, address, 'totalSupply');
+    multicaller.call(`${id}.virtualSupply`, address, 'getVirtualSupply');
+    multicaller.call(`${id}.actualSupply`, address, 'getActualSupply');
+    multicaller.call(`${id}.swapFee`, address, getSwapFeeFn(type));
+    multicaller.call(`${id}.rate`, address, 'getRate');
+    multicaller.call(`${id}.protocolYieldFeePercentageCache`, address, 'getProtocolFeePercentageCache', [2]);
+};
+
+const weightedCalls = (
+    { id, address }: PoolInput,
+    multicaller: Multicaller3
+) => {
+    multicaller.call(`${id}.weights`, address, 'getNormalizedWeights');
+};
+
+const lbpAndInvestmentCalls =(
+    { id, address }: PoolInput,
+    multicaller: Multicaller3
+) => {
+    multicaller.call(`${id}.weights`, address, 'getNormalizedWeights');
+    multicaller.call(`${id}.swapEnabled`, address, 'getSwapEnabled');
+};
+
+const linearCalls = (
+    { id, address }: PoolInput,
+    multicaller: Multicaller3
+) => {
+    multicaller.call(`${id}.targets`, address, 'getTargets');
+    multicaller.call(`${id}.wrappedTokenRate`, address, 'getWrappedTokenRate');
+};
+
+const stableCalls = (
+    { id, address, tokens }: PoolInput,
+    multicaller: Multicaller3
+) => {
+    multicaller.call(`${id}.amp`, address, 'getAmplificationParameter');
+
+    tokens.forEach(({ address: tokenAddress }, i) => {
+        multicaller.call(`${id}.tokenRate[${i}]`, address, 'getTokenRate', [tokenAddress]);
+        multicaller.call(`${id}.metaPriceRateCache[${i}]`, address, 'getPriceRateCache', [tokenAddress]);
+    });
+};
+
+const gyroECalls = (
+    { id, address }: PoolInput,
+    multicaller: Multicaller3
+) => {
+    multicaller.call(`${id}.tokenRates`, address, 'getTokenRates');
+};
+
+const poolTypeCalls = (type: PrismaPoolType, version = 1) => {
+    const do_nothing = () => ({});
+    switch (type) {
+        case 'WEIGHTED':
+            return weightedCalls;
+        case 'LIQUIDITY_BOOTSTRAPPING':
+        case 'INVESTMENT':
+            return lbpAndInvestmentCalls;
+        case 'STABLE':
+        case 'PHANTOM_STABLE':
+        case 'META_STABLE':
+            return stableCalls;
+        case 'GYROE':
+            if (version === 2) {
+            return gyroECalls;
+            } else {
+            return do_nothing;
+            }
+        case 'LINEAR':
+            return linearCalls;
+        default:
+            return do_nothing;
+    }
+};
+
+const parse = (result: OnchainData, decimalsLookup: { [address: string]: number }) => ({
+    amp: result.amp ? formatFixed(result.amp[0], String(result.amp[2]).length - 1) : undefined,
+    swapFee: formatEther(result.swapFee ?? '0'),
+    totalShares: formatEther(result.actualSupply || result.virtualSupply || result.totalSupply || '0'),
+    weights: result.weights?.map(formatEther),
+    targets: result.targets?.map(String),
+    poolTokens: result.poolTokens ? {
+        tokens: result.poolTokens[0].map((t) => t.toLocaleLowerCase()),
```

Comment:
I'm a fan of writing complete words for variables, we don't need to save any bytes :)